### ISSUE TITLE
Removing classes should not warn about references in classes to remove

### ISSUE THREAD

**Issue by Lin777** [guillep/pharo-core#4294]
If I want to remove a class A and a class B with B referencing A, currently, Calypso will warn me that A is still referenced before applying the refactoring.

IMO this should not happen because the class referencing A is part of the classes to remove. 

https://github.com/Lin777/Enjoliveur/issues/41

**Comment by bencoman** [comment]
What are the steps to reproduce? 
Please provide a script that can be run in a freshly downloaded image to:
1. create a condition where class B is referencing class A,
2. continue on produce the warning by trying to remove class A .

**Comment by Lin777** [comment]
**Steps for references warning:**
1. Create the classes: A, B where A refers to B.
2. Select classes A, B and delete them (Pressing Ctrl-X)
3. Produces a warning: There are 1 references to B, A. Show them?

**Steps for subclass warning:**
1. Create the classes: A, B where B is subclass of A
2. Select A, B and delete them.
3. Produces a warning: There are subclasses. Show them?

This should not happen since references and subclasses are also being deleted.

**Comment by bencoman** [comment]
Thanks Evelyn.  So to confirm, the code below is equivalent?...
```
Object subclass: #MyA
	instanceVariableNames: ''
	classVariableNames: ''
	package: 'Issue4294'.
	
Object subclass: #MyB
	instanceVariableNames: ''
	classVariableNames: ''
	package: 'Issue4294'.

#MyA asClass compile: 'reference
    ^ MyB'.

cmd := SycRemoveClassCommand new classes: {#MyA asClass. #MyB asClass}.
cmd execute
```

**Comment by Lin777** [comment]
Yes, it is equivalent and to test the subclasses error the script would be like this:

```
Object subclass: #MyA
	instanceVariableNames: ''
	classVariableNames: ''
	package: 'Issue4294'.
	
MyA subclass: #MyB
	instanceVariableNames: ''
	classVariableNames: ''
	package: 'Issue4294'.

cmd := SycRemoveClassCommand new classes: {#MyA asClass. #MyB asClass}.
cmd execute.
```

This will generate errors of the type RBRefactoringError, since it gets to execute the method primitiveExecute.

However, when executing from the UI, first execute this: aToolContext confirmUnusedClasses: classes from the method: #prepareFullExecutionInContext: which throws errors as alerts.

Therefore it performs a double validation and both already fix them in this PR: #4295
